routers/repo/view.go

```diff
@@ -264,14 +264,25 @@ func renderFile(ctx *context.Context, entry *git.TreeEntry, treeLink, rawLink st
 // Home render repository home page
 func Home(ctx *context.Context) {
 	if len(ctx.Repo.Repository.Units) > 0 {
-		tp := ctx.Repo.Repository.Units[0].Type
-		if tp == models.UnitTypeCode {
-			renderCode(ctx)
-			return
+		var firstAvailable models.UnitType
+		found := false
+		for _, unit := range ctx.Repo.Repository.Units {
+			if unit.Type == models.UnitTypeCode { // Check if unit Code is available
+				renderCode(ctx)
+				return
+			} else if !found && unit.Type != models.UnitTypeExternalTracker && unit.Type != models.UnitTypeExternalWiki { // Otherwise save first non external unit
+				firstAvailable = unit.Type
+				found = true
+			}
+		}
+
+		// If there is only external unit, just use first option
+		if !found {
+			firstAvailable = ctx.Repo.Repository.Units[0].Type
 		}
 
-		unit, ok := models.Units[tp]
-		if ok {
+		unit, ok := models.Units[firstAvailable]
+		if ok && unit.URI != models.UnitCode.URI {
```

Comment:
Why is the `unit.URI != models.UnitCodeURI` condition necessary? From what I can tell, there is no way `firstAvailable` can be `models.UnitTypeCode` at this point

Reply:
To avoid redirecting to the same page (causing an infinite loop). Safer method would be to check requested uri vs computed uri....

Reply:
@Morlinest but that can't happen as only unit with same url is code and if it will be allowed it will render and return and won't get to this point

Reply:
Actually I've just finished even more secure implementation. It can be used for each unit handler.